### ISSUE TITLE
Expand smoke tests to include reroute processors

### ISSUE THREAD

**Issue by marclop** [elastic/apm-server#14061]
## Description

While we have creation and upgrade smoke tests that are fairly comprehensive in the number of version that are covered, we've failed to increase the coverage of the test harness to include non-default but common. One very common scenario is using custom ingest pipelines to reroute documents based on source data fields.

## Test setup

We could use the reproduction details in #14060 to create a test where we assert that, errors, traces, spans and metrics are all correctly indexed before upgrading from previous minor to latest minor.

**Comment by raultorrecilla** [comment]
Moving this from it105 to it110

**Comment by endorama** [comment]
@ericywl where do we stand on this? Did https://github.com/elastic/apm-server/pull/15960 complete this?

**Comment by ericywl** [comment]
> where do we stand on this? Did https://github.com/elastic/apm-server/pull/15960 complete this?

Not really, I only added one test for reroute on 8.14 -> 8.16. I believe we should add this for all?

**Comment by ericywl** [comment]
We have implemented reroute tests in functional tests. Once that replaces existing smoke tests, we will have reroute tests for all version upgrades!
